pkg/archive/tar.go

```diff
@@ -76,10 +48,88 @@ func (e TarEntry) WriteTo(tw *tar.Writer) error {
 	return nil
 }
 
+type TarFile struct{ Files []*TarEntry }
+
+type TarArchiveStabilizer struct {
+	Name string
+	Func func(*TarFile)
+}
+
+type TarEntryStabilizer struct {
+	Name string
+	Func func(*TarEntry)
+}
+
+var AllTarStabilizers []any = []any{
+	StableTarFileOrder,
+	StableTarTime,
+	StableTarFileMode,
+	StableTarOwners,
+	StableTarXattrs,
+	StableTarDeviceNumber,
+}
+
+var StableTarFileOrder = TarArchiveStabilizer{
+	Name: "tar-file-order",
+	Func: func(f *TarFile) {
+		slices.SortFunc(f.Files, func(a, b *TarEntry) int {
+			return strings.Compare(a.Name, b.Name)
+		})
+	},
+}
+
+var StableTarTime = TarEntryStabilizer{
+	Name: "tar-time",
+	Func: func(e *TarEntry) {
+		e.ModTime = time.UnixMilli(0)
+		e.AccessTime = time.UnixMilli(0)
+		e.ChangeTime = time.Time{}
+		// NOTE: Without a PAX record, the tar library will disregard this value
+		// and write a USTAR-formatted file. Setting 'atime' ensures at least one
+		// record exists which will cause tar to serialize and re-parse it as PAX.
```

Comment:
I'm a little confused by the comment. The second half of the comment seems to describe that we must set atime as a way of forcing PAX but then why do we explicitly set FormatPAX? Also, it might be helpful to explain why we choose PAX instead of USTAR as the stable format.

Reply:
> we must set a time as a way of forcing PAX

Kind of. A PAX-format with no records IS a ustar-format archive. We use this strategy as a way of making working with the library more consistent I.e. format will actually be respected when we set it.

> why we choose PAX instead of USTAR as the stable format

The underlying blocks are identical between the two. PAX specifies how to provide additional records. So really, we _can't_ use ustar if there are, or we need, additional records (notably, required for precise timestamps and GNU tar compatibility).

Reply:
tweaked this to make this more clear